beancount_import/source/amazon.py

```diff
@@ -387,7 +395,7 @@ def make_amazon_transaction(
                         (INVOICE_DESCRIPTION, adjustment.description),
                     ]),
                 ))
-        if invoice.tax is not None and invoice.tax.number != ZERO:
+        if len(invoice.tax)>0 and invoice.tax.number != ZERO:
```

Comment:
What guarantees invoice.tax is not None?

Reply:
it relates to the above discussion: https://github.com/jbms/beancount-import/pull/144#discussion_r771871191

But I cannot recall why I removed the check on `None`...

For digital orders, `tax=[]` on order level anyway, so no need to check for `None`.

For regular orders, tax is set via:

```
tax = locale.parse_amount(
        get_field_in_table(payment_table, locale.regular_estimated_tax))
```
Which may return `None` if the field is not found...

Following up on the discussion above, I would vote for checking on `None` in `parse_regular_order_invoice` and setting `tax=[]` in this case. This ensures that in the resulting JSON `"tax": []`. Maybe also logging an error or warning might make sense, since so far all regular invoices contain tax information.

What are your thoughts on this? Did I miss something?

Reply:
So I tried to tackle this and stumbled upon an inconsistency in the current codebase:

According to the typedefs `tax: Amount` for `Order` and `Shipment`. But this is not correct. It is actually `tax: Sequence[Adjustment]` for `Shipment`. So to have an empty `tax` field on shipment level one needs to set `tax=[]`, whereas on order level `tax=None`, since `Amount` can be None.

I don't know why the typedef check does not catch this...

When it comes to digital orders, the current code sets `tax=[]` on order level. This is not correct, since `Amount` cannot be `[]`, it must be `None`. The code works nevertheless, since tax is always given on shipment level for digital orders and therefore tax on order level is not used anyway.

From the code perspective, I would go with the above corrections. But my dilemma arises when it comes to the JSON output for the tests: Setting `tax=None` translates into `tax: null` in the JSON. You suggested above that we should stick to the convention `field=[]`. I would drop this requirement since it would require additional modifications of the data (only for the test files...).

@Zburatorul are you fine with this? what are your thoughts?

Reply:
I added a proposal in [4be65b3](https://github.com/jbms/beancount-import/pull/144/commits/4be65b33d717c08cf13309efc51dc36c29066f8c)
- fix `shipment.tax` to type `List[Adjustment]`
- fix `order.tax` to type `Amount`
- results in `tax: null` in JSON for orders with no tax transaction (tax not given or included in item price)